src/noisepy/seis/application_modules/esyn_utils.py

```diff
@@ -0,0 +1,149 @@
+import math
+
+import numpy as np
+import pyasdf
+
+### -----
+# This scripts are aim to perform the 2-D radiative transfer equation
+# for scalar waves (Shang and Gao 1988; Sato 1993),
+# assuming of isotropic scattering and source radiation in infinite medium
+# to calculate synthesized energy densities Esyn.
+### -----
+
+
+### -----
+def read_pyasdf(sfile, ccomp):
+    # useful parameters from each asdf file
+    with pyasdf.ASDFDataSet(sfile, mode="r") as ds:
+        alist = ds.auxiliary_data.list()
+        try:
+            dt = ds.auxiliary_data[alist[0]][ccomp].parameters["dt"]
+            dist = ds.auxiliary_data[alist[0]][ccomp].parameters["dist"]
+            print("working on %s (comp: %s) that is %5.2fkm apart. dt: %.3f " % (sfile, ccomp, dist, dt))
+            # read stacked data
+            sdata = ds.auxiliary_data[alist[0]][ccomp].data[:]
+
+            # time domain variables
+            npts = sdata.size
+            tvec = np.arange(-npts // 2 + 1, npts // 2 + 1) * dt
+            return dist, dt, tvec, sdata
+
+        except Exception:
+            print("continue! no %s component exist" % ccomp)
+            return None
+
+
+### -----
+# Function that Calculate Mean Square
+def msValue(arr, n):
+    square = 0
+    mean = 0.0
+
+    # Calculate square
+    for i in range(0, n):
```

Comment:
Can this be done with numpy? https://numpy.org/doc/stable/reference/generated/numpy.square.html. Something like:
```
square = np.sum(np.square(arr))
```

Reply:
Updated!